src_c/_sdl2/video.pyx

```diff
@@ -686,6 +686,29 @@ cdef class Texture:
         if res < 0:
             raise error()
 
+    def update(self, surface):
+        # https://wiki.libsdl.org/SDL_UpdateTexture
+        # Should it accept a raw pixel data array too?
+        """ Update the texture with Surface.
+        This is a fairly slow function, intended for use with static textures that do not change often.
+
+        If the texture is intended to be updated often, 
+        it is preferred to create the texture as streaming and use the locking functions.
+
+        While this function will work with streaming textures, 
+        for optimization reasons you may not get the pixels back if you lock the texture afterward.
+
+        :param surface: source Surface.
+        """
+
+        if not pgSurface_Check(surface):
+            raise error("update source should be a Surface.")
+            
+        cdef SDL_Surface *surf = pgSurface_AsSurface(surface)
+
+        res = SDL_UpdateTexture(self._tex, &surf.clip_rect, surf.pixels, surf.pitch)
```

Comment:
Is `surf.clip_rect` used when blitting *to* `surf`? If so, it should probably use an argument instead

Reply:
Yea, should have an 'area' argument.